soc/xtensa/intel_adsp/common/soc.c

```diff
@@ -103,7 +103,7 @@ static __imr void power_init(void)
 }
 
 #if !DT_NODE_EXISTS(DT_NODELABEL(cavs0)) || CONFIG_MP_NUM_CPUS == 1
-void soc_idc_init(void) {}
+void soc_mp_init(void) {}
```

Comment:
FWIW: this one is closer to the root cause than the other patch.  I moved and renamed the function during the cleanup but missed this fallback stub.  Doesn't really matter, needs a little cleanup attention either way.